### PR TITLE
chore: promote crd to apiextensions.k8s.io/v1 and remove role assignments after e2e test

### LINKED ISSUE #661: Upgrade CRDs to apiextensions.k8s.io/v1
CRDs have been promoted to GA since Kubernetes 1.16:

https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/20180415-crds-to-ga.md

`aad-pod-identity` CRDs currently use apiextensions.k8s.io/v1beta1, which will be removed in the future: kubernetes/kubernetes#82022.

**Note:**

- The removal of `apiextensions.k8s.io/v1beta1` is planned for 1.22. So `aad-pod-identity` will continue to use `apiextensions.k8s.io/v1beta1` until the 1.21 release in order to support clusters < 1.16. Before the 1.22 release, we can make the switch to `apiextensions.k8s.io/v1` and set the default kube version of helm chart to 1.16+

**Alternatives:**

helm3 doesn't support templating in the crds dir. If this is possible, then we can determine the k8s version and use the appropriate `apiVersion`.

### LINKED ISSUE #994: ci: remove role assignments from aad-pod-identity-artifacts resource group
dangling role assignments are represented as `Identity not found` entry in the Azure portal
image
We will need to remove them after finish running the e2e test.

### PR DESCRIPTION
**Pull request by chewong**
Signed-off-by: Ernest Wong <chuwon@microsoft.com>

**Reason for Change**:

fixes #661
fixes #994 

- use controller-gen to generate YAML files for our CRDs
- existing users with 1.16+ clusters should not be affected by this PR since kube-apiserver already converts CRDs with apiVersion of v1beta1 to v1 automatically
- helm 3 does not re-configure the CRDs during an upgrade (meaning new fields like openAPIV3Schema won't be added), whereas `kubectl apply -f` does re-configure the CRDs to include new fields added in this PR
- existing CRDs data should be persisted unless the user explicitly runs `kubectl delete crds`

**Requirements**

- [x] squashed commits
- [x] included documentation
- [x] added unit tests and e2e tests (if applicable). See [test standard](https://github.com/Azure/aad-pod-identity/blob/master/CONTRIBUTING.md#test-standard) for more details.
- [x] ran `make precommit`

**Issue Fixed**:

**Please answer the following questions with yes/no**:

Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?

- [ ] yes
- [x] no

**Notes for Reviewers**:

### CODE REVIEW

File: test/e2e/backward_compat_test.go [hunk 1]
```diff
@@ -1,138 +0,0 @@
-// +build e2e
```
Comment (chewong) [inline test/e2e/backward_compat_test.go:1]: Removing this test due to v1 CRDs validation, which does not allow unknown fields to exist in the spec.

**Comment by chewong** [comment]
@aramase this PR is ready for review

File: manifest_staging/charts/aad-pod-identity/values.yaml [hunk 2]
```diff
@@ -232,7 +232,7 @@ nmi:
 
   # enable running aad-pod-identity on clusters with kubenet
   # default is false
-  allowNetworkPluginKubenet: false
+  allowNetworkPluginKubenet: true
```
Comment (aramase) [inline manifest_staging/charts/aad-pod-identity/values.yaml:235]: Was this an change intended?

File: pkg/apis/aadpodidentity/v1/types.go [hunk 3]
```diff
@@ -185,48 +187,51 @@ const (
...
 // AzureAssignedIdentitySpec contains the relationship
 // between an AzureIdentity and an AzureIdentityBinding.
 type AzureAssignedIdentitySpec struct {
 	metav1.ObjectMeta `json:"metadata,omitempty"`
-	AzureIdentityRef  *AzureIdentity        `json:"azureIdentityRef"`
-	AzureBindingRef   *AzureIdentityBinding `json:"azureBindingRef"`
-	Pod               string                `json:"pod"`
-	PodNamespace      string                `json:"podNamespace"`
-	NodeName          string                `json:"nodename"`
+	// +kubebuilder:validation:XEmbeddedResource
```
Comment (aramase) [inline pkg/apis/aadpodidentity/v1/types.go:206]: Could we also add a comment here on why we are adding this?

File: test/e2e/backward_compat_test.go (hunk 1 above)
Comment (aramase) [reply test/e2e/backward_compat_test.go:1]: sounds good!

File: manifest_staging/charts/aad-pod-identity/values.yaml (hunk 2 above)
Comment (chewong) [reply manifest_staging/charts/aad-pod-identity/values.yaml:235]: Good catch. I turned this on for testing but forgot to revert the change

**Review by aramase — APPROVED** [review]

**Pull request merged**